packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx

```diff
@@ -0,0 +1,211 @@
+interface KMSFields {
+  id: KMSProviderName;
+  title: string;
+  component: React.FC<{
+    autoEncryptionOptions: AutoEncryptionOptions;
+    updateConnectionFormField: UpdateConnectionFormField;
+    errors: ConnectionFormError[];
+  }>;
+}
+
+const options: KMSFields[] = [
+  {
+    title: 'Local KMS',
+    id: 'local',
+    component: LocalFields,
+  },
+  {
+    title: 'AWS',
+    id: 'aws',
+    component: AWSFields,
+  },
+  {
+    title: 'GCP',
+    id: 'gcp',
+    component: GCPFields,
+  },
+  {
+    title: 'Azure',
+    id: 'azure',
+    component: AzureFields,
+  },
+  {
+    title: 'KMIP',
+    id: 'kmip',
+    component: KMIPFields,
+  },
+];
+
+const containerStyles = css({
+  marginTop: spacing[3],
+});
+
+function CSFLETab({
+  connectionOptions,
+  updateConnectionFormField,
+  errors,
+}: {
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions: ConnectionOptions;
+}): React.ReactElement {
+  const [enabledKMSProviders, setEnabledKMSProviders] = useState<
+    KMSProviderName[]
+  >([]);
+  const autoEncryptionOptions =
+    connectionOptions.fleOptions?.autoEncryption ?? {};
+
+  errors = errorsByFieldTab(errors, 'csfle');
+
+  const handleFieldChanged = useCallback(
+    (
+      key: keyof AutoEncryptionOptions,
+      value?: AutoEncryptionOptions[keyof AutoEncryptionOptions]
+    ) => {
+      return updateConnectionFormField({
+        type: 'update-csfle-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  return (
+    <div className={containerStyles}>
+      <FormFieldContainer>
+        <TextInput
+          onChange={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            handleFieldChanged('keyVaultNamespace', value);
+          }}
+          name="csfle-keyvault"
+          data-testid="csfle-keyvault"
+          label="Key Vault Namespace"
+          type="text"
+          optional={false}
+          value={autoEncryptionOptions.keyVaultNamespace || ''}
+          errorMessage={errorMessageByFieldName(errors, 'keyVaultNamespace')}
+          state={
+            fieldNameHasError(errors, 'keyVaultNamespace') ? 'error' : 'none'
+          }
+          spellCheck={false}
+        />
+      </FormFieldContainer>
+      <EncryptedFieldConfigInput
+        // TOOD(COMPASS-5645): This says 'schemaMap', which is the
+        // FLE1 equivalent of the FLE2 'encryptedFieldConfig[Map?]'.
+        // Once 'encryptedFieldConfig' is available, we will start
+        // using it instead.
+        encryptedFieldConfig={autoEncryptionOptions?.schemaMap}
+        errorMessage={errorMessageByFieldName(errors, 'schemaMap')}
+        onChange={(value: Document | undefined) => {
+          handleFieldChanged('schemaMap', value);
+        }}
+      />
+      {/* TODO: Add Ace editor for EncryptedFieldConfig/SchemaMap */}
+      <Label htmlFor="TODO">KMS Providers</Label>
```

Comment:
These labels don't really look pretty right now, I should probably find something better for these